beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/DepositProvider.java

```diff
@@ -134,6 +135,16 @@ public void onNewFinalizedCheckpoint(
           return;
         }
         depositMerkleTree.finalize(finalizedState.getEth1Data(), heightOptional.get());
+        depositMerkleTree
+            .getSnapshot()
+            .ifPresent(
+                snapshot -> {
+                  LOG.trace("Storing DepositTreeSnapshot: {}", snapshot);
+                  final UpdatableStore.StoreTransaction storeTransaction =
+                      recentChainData.startStoreTransaction();
+                  storeTransaction.setFinalizedDepositSnapshot(snapshot);
+                  storeTransaction.commit().ifExceptionGetsHereRaiseABug();
```

Comment:
I don't think this should be going via the Store - it's not needed as part of the fork choice process and all our other eth1 storage stuff is already separate..  So this should be simplified to just directly call `StorageUpdateChannel` - make sure you pull in the latest changes from master and use the `storageUpdateChannel` in `BeaconChainController` so you get the right thing based on whether async IO is enabled or not.

Since it's still going through the same event channel as store transactions it's guaranteed that the updated finalized in the store will be recorded to disk before the new deposit snapshot we're storing here so there's no race conditions to worry about.